0000-template.md

```diff
@@ -1,74 +1,89 @@
 
 # Drawbacks
 [drawbacks]: #drawbacks
 
-Why should we *not* do this?
+This will require the compiler to be capable of performing checks for visual confusability on Unicode data.
 
 # Rationale and alternatives
 [alternatives]: #alternatives
 
-- Why is this design the best in the space of possible designs?
-- What other designs have been considered and what is the rationale for not choosing them?
-- What is the impact of not doing this?
+1. We could remove the `non_ascii_idents` feature entirely, and restrict identifiers to ASCII.
 
-# Prior art
-[prior-art]: #prior-art
+    This is the approach that Pony takes.  I do not like it because it weakens the Rust language’s internationalization support.
 
-Discuss prior art, both the good and the bad, in relation to this proposal.
-A few examples of what this can include are:
+2. We could stabilize the `non_ascii_idents` feature as-is.
 
-- For language, library, cargo, tools, and compiler proposals: Does this feature exist in other programming languages and what experience have their community had?
-- For community proposals: Is this done by some other community and what were their experiences with it?
-- For other teams: What lessons can we learn from what other communities have done here?
-- Papers: Are there any published papers or great posts that discuss this? If you have some relevant papers to refer to, this can serve as a more detailed theoretical background.
+    This renders Rust vulnerable to homograph attacks, which could be used to sneak code past code review.
 
-This section is intended to encourage you as an author to think about the lessons from other languages, provide readers of your RFC with a fuller picture.
-If there is no prior art, that is fine - your ideas are interesting to us whether they are brand new or if it is an adaptation from other languages.
+# Prior art
+[prior-art]: #prior-art
 
-Note that while precedent set by other languages is some motivation, it does not on its own motivate an RFC.
-Please also take into consideration that rust sometimes intentionally diverges from common language features.
+The motivation for the specific checks comes from a preprocessor for OCaml (whose name currently escapes me) that allows OCaml code to contain Unicode identifiers.
 
 # Unresolved questions
 [unresolved]: #unresolved-questions
 
-- What parts of the design do you expect to resolve through the RFC process before this gets merged?
-- What parts of the design do you expect to resolve through the implementation of this feature before stabilization?
-- What related issues do you consider out of scope for this RFC that could be addressed in the future independently of the solution that comes out of this RFC?
+- How can the confusability check be implemented?  We should have a fast path when there are *no* non-ASCII identifiers, but we also must be fast (at worst, O(N log N)) when there *are* non-ASCII identifiers.
```

Comment:
This needs to be the core of the detailed design, not an unresolved question, IMO.